applications/TrilinosApplication/custom_strategies/builder_and_solvers/trilinos_block_builder_and_solver.h

```diff
@@ -4,954 +4,842 @@
+                                     ModelPart& rModelPart,
+                                     TSystemMatrixType& A) override
+    {
+        KRATOS_ERROR << "Method BuildLHS_CompleteOnFreeRows not implemented in "
+                        "Trilinos Builder And Solver"
+                     << std::endl;
+    }
 
-
-    //**************************************************************************
-    //**************************************************************************
-    /** Solve the linear problem.
+    /**
+     * @brief This is a call to the linear system solver
+     * @param A The LHS matrix
+     * @param Dx The Unknowns vector
+     * @param b The RHS vector
      */
-    void SystemSolveWithPhysics(
-        TSystemMatrixType& A,
-        TSystemVectorType& Dx,
-        TSystemVectorType& b,
-        ModelPart& r_model_part
-    )
+    void SystemSolveWithPhysics(TSystemMatrixType& rA,
+                                TSystemVectorType& rDx,
+                                TSystemVectorType& rb,
+                                ModelPart& rModelPart)
     {
         KRATOS_TRY
 
         double norm_b;
-        if(TSparseSpace::Size(b) != 0)
-            norm_b = TSparseSpace::TwoNorm(b);
+        if (TSparseSpace::Size(rb) != 0)
+            norm_b = TSparseSpace::TwoNorm(rb);
         else
             norm_b = 0.00;
 
-        if(norm_b != 0.00)
-        {
-            if (this->GetEchoLevel()>1)
-                if (mrComm.MyPID() == 0) KRATOS_WATCH("entering in the solver");
-
-            if(BaseType::mpLinearSystemSolver->AdditionalPhysicalDataIsNeeded() )
-                BaseType::mpLinearSystemSolver->ProvideAdditionalData(A, Dx, b, BaseType::mDofSet, r_model_part);
-
-            if (this->GetEchoLevel()>3)
-            {
-                EpetraExt::RowMatrixToMatrixMarketFile( "A.mm", A, "matrixA", "lhs_matrix", true);
-                EpetraExt::MultiVectorToMatrixMarketFile( "b.mm", b, "vectorb","rhs_vector",true);
-                KRATOS_ERROR << "Stopping after printing the matrix";
-            }
-
-            if (this->GetEchoLevel()>3)
-            {
-                EpetraExt::RowMatrixToMatrixMarketFile( "A.mm", A, "matrixA", "block_matrix", true);
-                EpetraExt::MultiVectorToMatrixMarketFile( "b.mm", b, "vectorb","rhs_vector",true);
-                KRATOS_ERROR << "Stopping after printing the matrix";
-            }
+        if (norm_b != 0.00) {
+            if (BaseType::mpLinearSystemSolver->AdditionalPhysicalDataIsNeeded())
+                BaseType::mpLinearSystemSolver->ProvideAdditionalData(
+                    rA, rDx, rb, BaseType::mDofSet, rModelPart);
 
-            BaseType::mpLinearSystemSolver->Solve(A,Dx,b);
+            BaseType::mpLinearSystemSolver->Solve(rA, rDx, rb);
         }
-        else
-        {
-            TSparseSpace::SetToZero(Dx);
+        else {
+            TSparseSpace::SetToZero(rDx);
+            KRATOS_WARNING_ALL_RANKS(
```

Comment:
I would use the regular macro, there is no extra information to be gained here from printing on all ranks.